prov/tcp/src/tcpx_cq.c

```diff
@@ -48,11 +48,34 @@ static int tcpx_cq_close(struct fid *fid)
 	return 0;
 }
 
+static int tcpx_cq_control(struct fid *fid, int command, void *arg)
+{
+	struct util_cq *cq;
+	int ret;
+
+	cq = container_of(fid, struct util_cq, cq_fid.fid);
+
+	switch(command) {
+	case FI_GETWAIT:
+		if (!cq->wait)
+			return -FI_ENOSYS;
+
+		ret = cq->wait->wait_fid.fid.ops->control
+			(&cq->wait->wait_fid.fid, command, arg);
```

Comment:
Call through fi_control() rather than walking through the structures yourself.

Reply:
Changed. Thanks